beacon-chain/core/blocks/block_operations.go

```diff
@@ -929,9 +929,12 @@ func verifyTransfer(beaconState *pb.BeaconState, transfer *pb.Transfer, verifySi
 	if transfer.Amount > maxVal {
 		maxVal = transfer.Amount
 	}
+	if transfer.Amount+transfer.Fee > maxVal {
+		maxVal = transfer.Amount + transfer.Fee
```

Comment:
I don't feel good about this. Where's the test?

Why would we allow sender to initiate a transfer that exceeds the max val anyway? Shouldn't this be rejected?

Reply:
We dont allow a sender to initiate a transfer that exceeds maxVal, see L938 that's where we assert

this maxVal variable is used to check for over flow condition, we are not using it anywhere else

Reply:
Can you add a test for this check specifically?

Reply:
yep, added!